cmd/conduit/root/root.go

```diff
@@ -0,0 +1,120 @@
+	_ ecdysis.CommandWithSubCommands   = (*RootCommand)(nil)
+	_ ecdysis.CommandWithConfiguration = (*RootCommand)(nil)
+)
+
+type RootFlags struct {
+	Version bool `long:"version" short:"v" usage:"show current Conduit version" persistent:"true"`
+	conduit.Config
+}
+
+type RootCommand struct {
+	flags RootFlags
+	cfg   conduit.Config
+}
+
+func (c *RootCommand) Execute(_ context.Context) error {
+	if c.flags.Version {
+		_, _ = fmt.Fprintf(os.Stdout, "%s\n", conduit.Version(true))
+		return nil
+	}
+
+	e := &conduit.Entrypoint{}
+	e.Serve(c.cfg)
+	return nil
+}
+
+func (c *RootCommand) ParseConfig() ecdysis.Config {
+	path := filepath.Dir(c.flags.ConduitCfgPath)
+	c.cfg = conduit.DefaultConfigWithBasePath(path)
+
+	return ecdysis.Config{
+		EnvPrefix:  "CONDUIT",
+		ParsedCfg:  &c.cfg,
+		ConfigPath: c.flags.ConduitCfgPath,
+		DefaultCfg: c.cfg,
+	}
+}
+
+func (c *RootCommand) Usage() string { return "conduit" }
+
+func (c *RootCommand) Flags() []ecdysis.Flag {
+	flags := ecdysis.BuildFlags(&c.flags)
+
+	currentPath, err := os.Getwd()
+	if err != nil {
+		panic(cerrors.Errorf("failed to get current working directory: %w", err))
+	}
+
+	c.cfg = conduit.DefaultConfigWithBasePath(currentPath)
+	flags.SetDefault("config.path", c.cfg.ConduitCfgPath)
+	flags.SetDefault("db.type", c.cfg.DB.Type)
+	flags.SetDefault("db.badger.path", c.cfg.DB.Badger.Path)
+	flags.SetDefault("db.postgres.connection-string", c.cfg.DB.Postgres.ConnectionString)
+	flags.SetDefault("db.postgres.table", c.cfg.DB.Postgres.Table)
+	flags.SetDefault("db.sqlite.path", c.cfg.DB.SQLite.Path)
+	flags.SetDefault("db.sqlite.table", c.cfg.DB.SQLite.Table)
+	flags.SetDefault("api.enabled", c.cfg.API.Enabled)
+	flags.SetDefault("http.address", c.cfg.API.HTTP.Address)
+	flags.SetDefault("grpc.address", c.cfg.API.GRPC.Address)
+	flags.SetDefault("log.level", c.cfg.Log.Level)
+	flags.SetDefault("log.format", c.cfg.Log.Format)
+	flags.SetDefault("connectors.path", c.cfg.Connectors.Path)
+	flags.SetDefault("processors.path", c.cfg.Processors.Path)
+	flags.SetDefault("pipelines.path", c.cfg.Pipelines.Path)
+	flags.SetDefault("pipelines.exit-on-degraded", c.cfg.Pipelines.ExitOnDegraded)
+	flags.SetDefault("pipelines.error-recovery.min-delay", c.cfg.Pipelines.ErrorRecovery.MinDelay)
+	flags.SetDefault("pipelines.error-recovery.max-delay", c.cfg.Pipelines.ErrorRecovery.MaxDelay)
+	flags.SetDefault("pipelines.error-recovery.backoff-factor", c.cfg.Pipelines.ErrorRecovery.BackoffFactor)
+	flags.SetDefault("pipelines.error-recovery.max-retries", c.cfg.Pipelines.ErrorRecovery.MaxRetries)
+	flags.SetDefault("pipelines.error-recovery.max-retries-window", c.cfg.Pipelines.ErrorRecovery.MaxRetriesWindow)
+	flags.SetDefault("schema-registry.type", c.cfg.SchemaRegistry.Type)
+	flags.SetDefault("schema-registry.confluent.connection-string", c.cfg.SchemaRegistry.Confluent.ConnectionString)
+	flags.SetDefault("preview.pipeline-arch-v2", c.cfg.Preview.PipelineArchV2)
```

Comment:
As a future improvement, we could add something to ecdysis that sets default values from a struct. There would be less boilerplate, and we could make sure that the flag names match with tags.

Reply:
@hariso https://github.com/ConduitIO/ecdysis/issues/4